diff -up tigervnc-1.3.0/unix/xserver/hw/vnc/Input.cc.pointersync tigervnc-1.3.0/unix/xserver/hw/vnc/Input.cc
--- tigervnc-1.3.0/unix/xserver/hw/vnc/Input.cc.pointersync	2013-07-01 13:41:24.000000000 +0100
+++ tigervnc-1.3.0/unix/xserver/hw/vnc/Input.cc	2014-05-22 11:47:16.707816457 +0100
@@ -197,6 +197,14 @@ void InputDevice::PointerMove(const rfb:
 
 void InputDevice::PointerSync(void)
 {
+	if (initialized) {
+		int x, y;
+
+		GetSpritePosition (pointerDev, &x, &y);
+		cursorPos.x = x;
+		cursorPos.y = y;
+	}
+
 	if (cursorPos.equals(oldCursorPos))
 		return;
 
