diff -up tigervnc-1.0.90-20100420svn4030/unix/xserver/hw/vnc/Input.cc.rh586406 tigervnc-1.0.90-20100420svn4030/unix/xserver/hw/vnc/Input.cc
--- tigervnc-1.0.90-20100420svn4030/unix/xserver/hw/vnc/Input.cc.rh586406	2010-04-14 11:24:06.000000000 +0200
+++ tigervnc-1.0.90-20100420svn4030/unix/xserver/hw/vnc/Input.cc	2010-05-13 12:23:14.418190525 +0200
@@ -31,8 +31,10 @@ extern "C" {
 #define class c_class
 #include "inputstr.h"
 #include "mi.h"
-#ifndef XKB_IN_SERVER
-#define XKB_IN_SERVER
+#if XORG < 18
+  #ifndef XKB_IN_SERVER
+    #define XKB_IN_SERVER
+  #endif
 #endif
 #ifdef XKB
 /*
@@ -712,7 +714,12 @@ static KeySym KeyCodetoKeySym(KeySymsPtr
 	}
 
 	if ((per <= (col|1)) || (syms[col|1] == NoSymbol)) {
-		XConvertCase(syms[col&~1], &lsym, &usym);
+#if XORG >= 18
+		XkbConvertCase
+#else
+		XConvertCase
+#endif
+			    (syms[col&~1], &lsym, &usym);
 		if (!(col & 1))
 			return lsym;
 		/*
