unchanged:
--- unix/vncviewer/vncviewer.man	(revision 3902)
+++ unix/vncviewer/vncviewer.man	(working copy)
@@ -156,7 +156,9 @@
 .B \-LowColorLevel, \-LowColourLevel \fIlevel\fP
 Selects the reduced color level to use on slow links.  \fIlevel\fP can range
 from 0 to 2, 0 meaning 8 colors, 1 meaning 64 colors (the default), 2 meaning
-256 colors.
+256 colors. Note that decision if reduced color level is used is made by
+vncviewer. If you would like to force vncviewer to use reduced color level
+use \fB-AutoSelect=0\fP parameter.
 
 .TP
 .B \-PreferredEncoding \fIencoding\fP
only in patch2:
unchanged:
--- unix/vncviewer/vncviewer.cxx	(revision 3902)
+++ unix/vncviewer/vncviewer.cxx	(working copy)
@@ -337,6 +337,13 @@
 			&& !::fullColour.hasBeenSet()
 			&& !::fullColourAlias.hasBeenSet());
   }
+  if (!::fullColour.hasBeenSet() && !::fullColourAlias.hasBeenSet()) {
+    // Default to FullColor=0 if AutoSelect=0 && LowColorLevel is set
+    if (!::autoSelect && (::lowColourLevel.hasBeenSet() ||
+			  ::lowColourLevelAlias.hasBeenSet())) {
+      ::fullColour.setParam(false);
+    }
+  }
   if (!::customCompressLevel.hasBeenSet()) {
     // Default to CustomCompressLevel=1 if CompressLevel is used.
     ::customCompressLevel.setParam(::compressLevel.hasBeenSet());
